src/Refactoring-Tests-Core/RBNamespaceTest.class.st

```diff
@@ -26,7 +26,64 @@ RBNamespaceTest >> testAllClassesDo [
 	self assert: classes equals: 0
 ]
 
-{ #category : #'class tests' }
+{ #category : #'tests - on environment' }
+RBNamespaceTest >> testAllImplementorsOnClassSubset [
+
+	| rbclasses rbNamespace2 |
+	rbclasses := RBClassEnvironment classes: {
+			             RBClassToRename.
+			             RBSubclassOfClassToRename }.
+	rbNamespace2 := RBNamespace onEnvironment: rbclasses.
+	"there are several method #method1 in the system will in this slice we only have 1."
+	self
+		assert: (rbNamespace2 allImplementorsOf: #method1) size
+		equals: 1.
+
+	self
+		assert: (rbNamespace2 allImplementorsOf: #printOn:) size
+		equals: 1.
+	self
+		assert:
+		(rbNamespace2 allImplementorsOf: #equallyDefinedInSubclass) size
+		equals: 2
+]
+
+{ #category : #'tests - on environment' }
+RBNamespaceTest >> testAllReferencesToOnClassSubset [
+
+	| rbclasses rbNamespace2 |
+	rbclasses := RBClassEnvironment classes: {
+			             RBClassToRename.
+			             RBSubclassOfClassToRename }.
+	rbNamespace2 := RBNamespace onEnvironment: rbclasses.
+
```

Comment:
Just a side note. Since a few versions of Pharo we have #assertEmpty: and #denyEmpty: which give better logs on the CI because in case of failure they print the content of the collection. It helps a lot to debug the failures